sdk/identity/Azure.Identity/src/Credentials/AzurePipelinesServiceConnectionCredentialOptions.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Options used to configure the <see cref="AzurePipelinesServiceConnectionCredential"/>.
+    /// </summary>
+    public class AzurePipelinesServiceConnectionCredentialOptions : TokenCredentialOptions, ISupportsDisableInstanceDiscovery, ISupportsAdditionallyAllowedTenants, ISupportsTokenCachePersistenceOptions
+    {
+        internal CredentialPipeline Pipeline { get; set; }
+
+        internal MsalConfidentialClient MsalClient { get; set; }
+
+        /// <summary>
+        /// The URI of the TFS collection or Azure DevOps organization.
+        /// </summary>
+        public string CollectionUri { get; set; } = Environment.GetEnvironmentVariable("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
+
+        /// <summary>
+        /// The type of build to get the token for
+        /// </summary>
+        public string HubName { get; set; } = Environment.GetEnvironmentVariable("SYSTEM_HOSTTYPE");
+
+        /// <summary>
+        /// A unique identifier for a single attempt of a single job. The value is unique to the current pipeline.
+        /// </summary>
+        public string JobId { get; set; } = Environment.GetEnvironmentVariable("SYSTEM_JOBID");
+
+        /// <summary>
+        /// A string-based identifier for a single pipeline run.
+        /// </summary>
+        public string PlanId { get; set; } = Environment.GetEnvironmentVariable("SYSTEM_PLANID");
+
+        /// <summary>
+        /// The security token used by the running build.
+        /// </summary>
+        public string SystemAccessToken { get; set; } = Environment.GetEnvironmentVariable("SYSTEM_ACCESSTOKEN");
```

Comment:
the value of this will be a secret right?

Reply:
Just curious if calling it `public` has any impact on it being a secret?

Reply:
Good question - I'll consider removing it as an option and instead load directly from the env.

Reply:
I decided to keep the property but make the getter `internal` to maintain testability